aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -394,35 +423,32 @@ SparseTensor& sparse_mask_out_cpu(SparseTensor& r, const Tensor& t, const Sparse
     return r;
   }
 
-  // NB: Relies on mask._nnz() == 0 test above
-  auto mask_indices_accessor = mask_indices.accessor<int64_t, 2>();
-
   if (dim > sparse_dim) {
-    // NB: This used to reuse buffers, but I deoptimized it
-    for (int64_t i = 0; i < r_nnz; i++) {
-      Tensor srcBuffer = t;
-      for (int64_t d = 0; d < sparse_dim; d++) {
-        srcBuffer = srcBuffer.select(0, mask_indices_accessor[d][i]);
-      }
-      Tensor dstBuffer = r_values.select(0, i);
-      dstBuffer.copy_(srcBuffer);
+    LongTensor indices = at::zeros({mask._nnz()}, mask_indices.options());
+
+    for (int64_t d = 0; d < mask.sparse_dim(); d++) {
```

Comment:
This loop could probably be replaced by https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/SparseTensorUtils.h#L84

Reply:
Thanks! Indeed this can be replaced by `flatten_indices()`, but for some reason it gets slower after the replacement (1.38 ms ± 179 µs after replacement vs 594 µs ± 68.9 µs before it). I will keep the current implementation and note down what the current code is trying to do

Reply:
Interesting. Even for CUDA? If it is always faster, we should replace `flatten_indices` with this.

Reply:
not needed for this PR tho